### PR TITLE
Implement generators

### LINKED ISSUE #390: cli generator updates
1. Generated plugins should default to MIT license.  That will maximize utilization and not cause conflicts with Enterprise Edition.
2. Consider including generators for conversions, limits responses, widgets, microservices, and tools (perhaps just cloning an existing template repo?)
3. rake build VERSION=1.0.0 should not put timestamp in filename

### PR DESCRIPTION
**Pull request by jmthomas**
closes #390

### CODE REVIEW

**Comment by jmthomas** [comment]
@ryanmelt Not finished but I wanted to get your take on it. I think it makes sense to use the template system I already had so we can rename things and make it nicer for the end user. I need to find a good way to test these.

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          docker image ls | grep localhost:12345/openc3/openc3-redis | grep mine
+          docker image ls | grep localhost:12345/openc3/openc3-traefik | grep mine
+      # - name: Build plugin
+      #   shell: 'script -q -e -c "bash {0}"'
+      #   run: |
+      #     ./openc3.sh cliroot generate plugin PW_TEST
+      #     cd openc3-cosmos-pw-test
+      #     ../openc3.sh cliroot rake build VERSION=1.0.0
+      #     sudo mv openc3-cosmos-pw-test*.gem openc3-cosmos-pw-test-1.0.0.gem
+      #     ../openc3.sh cliroot validate openc3-cosmos-pw-test-1.0.0.gem
```
Comment (ryanmelt) [inline .github/workflows/cli.yml:98]: why is this commented out?

File: openc3/templates/microservice/microservices/TEMPLATE/template.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        # Do your microservice work here
+        Logger.info("Template Microservice ran")
+        # cmd("INST ABORT")
+
+        run_time = Time.now - start_time
+        delta = @period - run_time
+        if delta > 0
+          # Delay till the next period
+          break if @sleeper.sleep(delta) # returns true and breaks loop on shutdown
+        end
```
Comment (ryanmelt) [inline openc3/templates/microservice/microservices/TEMPLATE/template.rb:46]: Run should increment @count

**Review by ryanmelt — CHANGES_REQUESTED** [review]
Also refactor to make base plugin and be able to add targets.

**Review by ryanmelt — APPROVED** [review]

**Pull request merged**